packages/vite/src/node/constants.ts

```diff
@@ -15,6 +15,18 @@ export const DEFAULT_EXTENSIONS = [
   '.json'
 ]
 
+// file extensions that will be transformed to js modules
+export const KNOWN_JS_SRC_EXTENSIONS = new Set([
+  '.js',
+  '.jsx',
+  '.ts',
+  '.tsx',
+  '.mjs',
+  '.vue',
+  '.marko',
+  '.svelte'
```

Comment:
When we published this feature we could contact the `vue`, `marko` and `svelte` plugin owners and ask for that they add their extensions via their plugins
Then later on we should remove these framework extensions from Vite core

Reply:
Yes, ultimately that is the goal. For now i kept them because it requires plugin support and users updating said plugins. Maybe they should be marked as deprecated and removed with the next major only. 
As this hasn't been widely used and if we coordinate it correctly with the plugins removal can happen sooner :thinking:

Reply:
We are going to release a minor version in around 10 days, we may be able to do it there. I think maybe we could deprecate them in v2.3 and wait til v2.4 to remove them.

Reply:
@patak-js let the plugin-owners and the eco-system catch up and remove them in e.g. vite 2.6+